interactions/ext/hybrid_commands/context.py

```diff
@@ -175,16 +175,23 @@ def typing(self) -> Typing | DeferTyping:
             return DeferTyping(self._slash_ctx, self.ephemeral)
         return self.channel.typing
 
-    async def defer(self, ephemeral: bool = False) -> None:
+    async def defer(self, ephemeral: bool = False, suppress_error: bool = False) -> None:
         """
         Either defers the response (if used in an interaction) or triggers a typing indicator for 10 seconds (if used for messages).
 
+        ???+ note "Interaction Note"
+            This method's ephemeral settings override the ephemeral settings of `send()`.
+
+            For example, deferring with `ephemeral=True` will make the interaction response ephemeral even with
+            `send(ephemeral=False)`.
+
         Args:
             ephemeral: Should the response be ephemeral? Only applies to responses for interactions.
+            suppress_error: Should errors on deferring be suppressed than raised.
 
         """
         if self._slash_ctx:
-            await self._slash_ctx.defer(ephemeral=ephemeral)
+            await self._slash_ctx.defer(ephemeral=ephemeral, suppress_error=suppress_error)
         else:
             await self.channel.trigger_typing()
```

Comment:
To be consistent, you probably should also suppress errors for the `trigger_typing` variant.

Reply:
I didn't think of adding one because `trigger_typing()` doesn't have the sort of errors expected by an extra call to `defer()`.

That said, I added suppression (on only `HTTPException`) near that snippet of code.